pallets/parachain-staking/src/mock.rs

```diff
@@ -149,8 +149,8 @@ pub(crate) struct ExtBuilder {
 	balances: Vec<(AccountId, Balance)>,
 	// [collator, amount]
 	collators: Vec<(AccountId, Balance)>,
-	// [delegator, collator, delegation_amount]
-	delegations: Vec<(AccountId, AccountId, Balance)>,
+	// [delegator, collator, delegation_amount, auto_compound_percent]
+	delegations: Vec<(AccountId, AccountId, Balance, Percent)>,
```

Comment:
Was `Percent` chosen to reduce the size as much as absolutely possible? It has extremely low precision as it is a `u8`.

Reply:
This comes from how Percent is being stored https://docs.rs/sp-runtime/latest/sp_runtime/struct.Percent.html#method.deconstruct

Reply:
Right, it's stored as a `u8` but that also means it has very poor precision as a percentage. I'm just making sure the trade off is intentional.

Reply:
It would be up for discussion, I merely went with a simplified approach unless otherwise required. Just wondering if someone would actually want a `20.50%` autocompound. If someone has different opinions, we can switch to `Perbill` or something I guess.

Reply:
If we don't care about the storage size, we might as well include more precision IMO.

The "options" are here: https://crates.parity.io/src/sp_arithmetic/per_things.rs.html#1404

Summary of options:
```
Percent: u8, 0 - 100 used range
PerU16: u16, 0 - 65_535 used range
Permill: u32, 0 - 1_000_000 used range
Perbill: u32, 0 - 1_000_000_000 used range
Perquintill: u64, 0 - 1_000_000_000_000_000_000 used range
```

My suggestion would be `PerU16` to keep size down (2 bytes) or `Perbill` otherwise (4 bytes). `Perquintill` seems quite overkill...

Reply:
One consideration would be how the users and smart contracts would provide this value (also to precompiles) and what kind of return values we send to precompiles. Here `u8` is the most intuitive, maybe @nanocryk knows what's the best way to send (and return) percentages represented as PerX and if it's nice for SCs.

Reply:
I think SC devs are used to extremely high precision (they work in increments of Wei all the time :slightly_smiling_face:) But I do agree that `Percent` is inherently very user-friendly...